src/components/SettlementButton.tsx

```diff
@@ -197,7 +197,7 @@ function SettlementButton({
         }
 
         if (iouPaymentType === CONST.IOU.REPORT_ACTION_TYPE.APPROVE) {
-            IOU.approveMoneyRequest(iouReport);
+            IOU.approveMoneyRequest(iouReport ?? {});
```

Comment:
@ZhenjaHorbach @mountiny 
shouldn't this be moved to the if check? Are we sure we want to pass in an empty object to that function when the report action type is `approved` and I would assume - ready to process? Seems some of the keys would end up like `report_undefined` etc.

Reply:
well imo if that can happen the function should handle such case.

I am not familiar with this part of the code, just stumbled upon that today when we were fixing the latest main merge on a different pr so I am not sure if not calling `IOU.approveMoneyRequest` in such case is valid, just applying some logic here, but it might be flawed as stated above.